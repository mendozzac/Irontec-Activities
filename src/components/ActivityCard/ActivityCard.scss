@import "../../styles/colors";

h2 {
  margin-left: 0;
}

li {
  list-style: none;
}

.data-card {
  width: 250px;
  height: 220px;
  padding: 20px;
  display: flex;
  flex-direction: column;
  border-radius: 10px;
  overflow: hidden;
  background-image: linear-gradient(to bottom right, #ec4040, #f59494);
  color: #fff;
  box-shadow: $shadow;
}

span {
  margin-bottom: 5px;
}

.data-card_title {
  font-weight: bold;
}
