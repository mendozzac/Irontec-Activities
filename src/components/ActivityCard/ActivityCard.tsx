import { Activity } from "../../types";
import "./ActivityCard.scss";

interface Card {
  activity: Activity;
}

const ActivityCard: React.FC<Card> = (activity) => {
  return (
    <>
      <li>
        <div className="data-card">
          {activity ? (
            <>
              <span className="data-card_title">
                {activity.activity.activity}
              </span>
              <span>{activity.activity.type}</span>
              <span>{activity.activity.participants}</span>
              <span>{activity.activity.price}</span>
              <span>{activity.activity.link}</span>
              <span>{activity.activity.key}</span>
              <span>{activity.activity.accessibility}</span>
            </>
          ) : (
            <span>hola</span>
          )}
        </div>
      </li>
    </>
  );
};

export default ActivityCard;
