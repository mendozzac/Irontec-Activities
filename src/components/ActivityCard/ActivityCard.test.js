import { render, screen } from "@testing-library/react";
import ActivityCard from "./ActivityCard";
import TestRenderer from "react-test-renderer";

describe("Given an ActivityCard component", () => {
  describe("When it receives activity information", () => {
    test("Then it should render a card with the information of the activity", () => {
      const activity = {
        accessibility: 0.25,
        activity: "Make a simple musical instrument",
        key: "7091374",
        link: "",
        participants: 1,
        price: 0.4,
        type: "music",
      };

      render(<ActivityCard activity={activity} key={activity.key} />);
      const newActivity = screen.getByText("music");

      expect(newActivity).toBeInTheDocument();
    });
  });
  describe("When it receives an activity", () => {
    test("Then it should render a card with the activity", () => {
      const newActivity = {};

      const activityCard = TestRenderer.create(
        <ActivityCard activity={newActivity} key={newActivity.key} />
      );

      expect(activityCard.toJSON()).toMatchSnapshot();
    });
  });
});
