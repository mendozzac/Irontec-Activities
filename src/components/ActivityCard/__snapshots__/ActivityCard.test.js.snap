// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an ActivityCard component When it receives an activity Then it should render a card with the activity 1`] = `
<li>
  <div
    className="data-card"
  >
    <span
      className="data-card_title"
    />
    <span />
    <span />
    <span />
    <span />
    <span />
    <span />
  </div>
</li>
`;
